app/src/main/java/de/danoeh/antennapod/fragment/AudioPlayerFragment.java

```diff
@@ -551,6 +555,13 @@ public boolean onMenuItemClick(MenuItem item) {
                 return true;
             case R.id.audio_controls:
                 PlaybackControlsDialog dialog = PlaybackControlsDialog.newInstance();
+                dialog.setOnRepeatChanged(repeat -> {
```

Comment:
This is not reliable when the PlaybackControlsDialog fragment is recreated by the Android system. Please listen to the preferences changes directly instead.

Reply:
Sorry, but what do you mean by 'directly'?

Reply:
Using a Consumer still breaks when the fragment instance is recreated. Instead, you can subscribe to preference changes. The dialog then no longer needs to notify its caller.

```
PreferenceManager.getDefaultSharedPreferences(context).registerOnSharedPreferenceChangeListener(this);

    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        if (UserPreferences.PREF_REPEAT_EPISODE.equals(key)) {
            // update icon state
        }
    }
```